### PR TITLE
Enable `PsesInternalHostTests` (previously `PowerShellContextTests`)

### PR DESCRIPTION
**Pull request by andschwa**
I was fortunately able to re-enable the `PowerShellContextService` tests with the new `PsesInternalHost` pretty easily.

Also fixes https://github.com/PowerShell/vscode-powershell/issues/3677.

### CODE REVIEW

**Comment by andschwa** [comment]
Oops, gotta fix it for Windows 🤷

**Comment by andschwa** [comment]
All fixed!

**Comment by andschwa** [comment]
Hey @JustinGrote I covered #1629 with a regression test and cherry-picked the commit. Can you review this? We can close the other PR.

File: src/PowerShellEditorServices/Services/PowerShell/Host/PsesInternalHost.cs [hunk 1]
```diff
@@ -29,11 +29,11 @@ internal class PsesInternalHost : PSHost, IHostSupportsInteractiveSession, IRuns
     {
         private const string DefaultPrompt = "PSIC> ";
         // This is a default that can be overriden at runtime by the user or tests.
-        private static string s_bundledModulePath = Path.GetFullPath(Path.Combine(
+        private static string _bundledModulePath = Path.GetFullPath(Path.Combine(
```
Comment (SeeminglyScience) [inline src/PowerShellEditorServices/Services/PowerShell/Host/PsesInternalHost.cs:32]: Any particular reason for this? I usually stick with [dotnet/runtime's style guide](https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/coding-style.md). I believe that's what PowerShell/PowerShell uses as well.

**Review by SeeminglyScience — COMMENTED** [review]
LGTM! + one question

File: src/PowerShellEditorServices/Services/PowerShell/Host/PsesInternalHost.cs (hunk 1 above)
Comment (andschwa) [reply src/PowerShellEditorServices/Services/PowerShell/Host/PsesInternalHost.cs:32]: Roslyn Analyzer was complaining, but then it kept complaining!

Comment (andschwa) [reply src/PowerShellEditorServices/Services/PowerShell/Host/PsesInternalHost.cs]: Reverted.

**Review by SeeminglyScience — APPROVED** [review]
LGTM!

**Comment by andschwa** [comment]
Part of #1677

**Pull request merged**